src/fundus/scraping/html.py

```diff
@@ -274,99 +37,188 @@ class HTML:
-                yield None
                 continue
 
-            session = await session_handler.get_session()
+            session = session_handler.get_session()
 
             try:
-                async with session.get(url, headers=self.request_header) as response:
-                    if filter_url(str(response.url)):
-                        basic_logger.debug(f"Skipped responded URL '{str(response.url)}' because of URL filter")
-                        yield None
-                        continue
-                    html = await response.text()
-                    response.raise_for_status()
-
-            except (HTTPError, ClientError, HttpProcessingError, UnicodeError) as error:
+                response = session.get(url, headers=self.request_header)
+
+            except (HTTPError, ConnectionError) as error:
                 basic_logger.info(f"Skipped requested URL '{url}' because of '{error}'")
-                yield None
+                if isinstance(error, HTTPError) and error.response.status_code >= 500:
+                    basic_logger.info(f"Skipped {self.publisher} due to server errors: '{error}'")
                 continue
 
             except Exception as error:
                 basic_logger.warning(f"Warning! Skipped  requested URL '{url}' because of an unexpected error {error}")
-                yield None
                 continue
 
-            if response.history:
-                basic_logger.info(f"Got redirected {len(response.history)} time(s) from {url} -> {response.url}")
-
-            source = (
-                WebSource(self.publisher, type(self.url_source).__name__, self.url_source.url)
-                if isinstance(self.url_source, URLSource)
-                else HTMLSource(self.publisher)
-            )
-
-            yield HTML(
-                requested_url=url,
-                responded_url=str(response.url),
-                content=html,
-                crawl_date=datetime.now(),
-                source=source,
-            )
+            else:
+                if filter_url(str(response.url)):
+                    basic_logger.debug(f"Skipped responded URL '{str(response.url)}' because of URL filter")
+                    continue
+                html = response.text
+
+                if response.history:
+                    basic_logger.info(f"Got redirected {len(response.history)} time(s) from {url} -> {response.url}")
+
+                source = (
+                    WebSourceInfo(self.publisher, type(self.url_source).__name__, self.url_source.url)
+                    if isinstance(self.url_source, URLSource)
+                    else SourceInfo(self.publisher)
+                )
+
+                yield HTML(
+                    requested_url=url,
+                    responded_url=str(response.url),
+                    content=html,
+                    crawl_date=datetime.now(),
+                    source=source,
+                )
```

Comment:
Since each source now has a corresponding SourceInfo class, I think it would make sense to change the associated variables and parameters source to `source_info` where a SourceInfo is used. Otherwise these two may get mixed up.

Reply:
Maybe this would be better for a separate PR.

Reply:
Good point, I implemented the changes within this PR.

Reply:
Looks good!